inclusive-design/wecount.inclusivedesign.ca
# fix: solve Initiatives tags sorting and duplicate tag issues

## Description

This update fixes two issues with the Initiatives (formerly Views) page. The first is #816, and the proposed solution was to make all tags lowercase only. The other issue is an undocumented bug which fails builds if two articles had the same tag with different case usage (e.g. `"accessibility"` and `"Accessibility"`).

Seeing that the solution in both cases was to consolidate tags to a single spelling, these are both addressed with a single pull request.

## Steps to test

1. Run the site locally
2. Go to the Initiatives page: http://localhost:3000/views/
3. Expand the "Search Filters" panel
4. Take note of the listed tags.
   Expected state: no duplicate spellings, sort order is alphabetical going left-to-right then top-to-bottom, uppercase and lowercase have no bearing on sort order.
5. Select the "Inclusion" checkbox and activate "Apply Filters"
6. After the matching results are shown (expected: 3 results), click on the link for "We Count Evaluates Toronto's Quayside ..."
7. On the next page, scroll down and click on the link for the "Inclusion" tag
8. On the page for the "Inclusion" tag, there should be six results including ones titled "A Future Date Conference" and "The Walrus Talks: Inclusion" which are news articles.

## Additional information

The initially-proposed fix of making all tags lowercase-only made the multiple acronym tags appear somewhat awkward and more difficult to identify visually. It appeared that getting the site to work by accepting the "lowest" spelling of a given tag rather than enforcing all of them to be lowercase-only would make things significantly more palatable, so this is the approach I took.

Considerations and brief attempts were also made towards better factoring for the fix, including more efficient algorithms and data structures to iterate over the various tags collections, however the time I estimated it would take to design and implement these would far exceed the additional build times created by the current factoring.

## Related issues

Resolves #816
